packages/adapters/src/adapters/Cornerstone3D/Segmentation/generateSegmentation.ts

```diff
@@ -30,36 +54,66 @@ function generateSegmentation(images, labelmaps, metadata, options = {}) {
  * @returns The Seg derived dataSet.
  */
 function _createMultiframeSegmentationFromReferencedImages(
-    images,
-    metadata,
-    options
+  images,
+  metadata,
+  options
 ) {
-    const datasets = images.map(image => {
-        // add the sopClassUID to the dataset
-        const instance = metadata.get("instance", image.imageId);
-        return {
-            ...image,
-            ...instance,
-            // Todo: move to dcmjs tag style
-            SOPClassUID: instance.SopClassUID || instance.SOPClassUID,
-            SOPInstanceUID: instance.SopInstanceUID || instance.SOPInstanceUID,
-            PixelData: image.voxelManager.getScalarData(),
-            _vrMap: {
-                PixelData: "OW"
-            },
-            _meta: {}
-        };
-    });
+  const studyData = metadata.get(MetadataModules.STUDY_DATA, images[0].imageId);
+  const datasets = images.map((image) => {
+    const { imageId } = image;
+    const seriesData = metadata.get(MetadataModules.SERIES_DATA, imageId);
+    const imageData = metadata.get(MetadataModules.IMAGE_DATA, imageId);
+    return {
```

Comment:
This type of image reference can be created against any of the metadata providers, allowing creation of segmentation against the wado uri and dicom part 10 providers as well.